samza-core/src/main/java/org/apache/samza/operators/MessageStreamImpl.java

```diff
@@ -101,13 +108,16 @@ public void sink(SinkFunction<M> sinkFn) {
     return wndOp.getNextStream();
   }
 
-  @Override public <K, OM, RM> MessageStream<RM> join(MessageStream<OM> otherStream, JoinFunction<K, M, OM, RM> joinFn) {
+  @Override
+  public <K, JM, RM> MessageStream<RM> join(MessageStream<JM> otherStream, JoinFunction<K, M, JM, RM> joinFn, Duration ttl) {
     MessageStreamImpl<RM> outputStream = new MessageStreamImpl<>(this.graph);
 
-    PartialJoinFunction<K, M, OM, RM> parJoin1 = new PartialJoinFunction<K, M, OM, RM>() {
+    PartialJoinFunction<K, M, JM, RM> thisPartialJoinFn = new PartialJoinFunction<K, M, JM, RM>() {
```

Comment:
Minor preference for PartialJoinFunction as well. If we change, we should probably call it PartialJoinManager instead of PartialJoinState.